index.js

```diff
@@ -65,7 +67,11 @@ exports.parseQuery = function parseQuery(query) {
 	};
 	if(!query) return {};
 	if(typeof query !== "string")
-		return query;
+		return (util.deprecate(function () {},
+		"loaderUtils.parseQuery() just received a value type of " + typeof query +
+		" which can be problematic, see https://github.com/webpack/loader-utils/issues/56" + os.EOL +
+		"parseQuery() will be replaced with getOptions() in the next major version of loader-utils."))(),
+		query;
```

Comment:
Doesn't using deprecate this way will report the message on every call instead of only once?